Clockwise.Redis/CircuitBreakerStoragePartition.cs

```diff
@@ -0,0 +1,164 @@
+                    JsonConvert.SerializeObject(newState, JsonSerializationSettings));
+            }
+
+            return TryDeserialise(lastSerialisedState);
+        }
+
+        private static CircuitBreakerStateDescriptor TryDeserialise(string serialised)
+        {
+            return IsNullOrWhiteSpace(serialised)
+                ? null
+                : JsonConvert.DeserializeObject<CircuitBreakerStateDescriptor>(serialised, JsonSerializationSettings);
+        }
+        private async Task TransitionStateTo(CircuitBreakerStateDescriptor targetState, TimeSpan? expiry = null)
+        {
+            var serialsied = JsonConvert.SerializeObject(targetState, JsonSerializationSettings);
+            var stateExpiry = targetState.State == CircuitBreakerState.Open && expiry == null
+                ? TimeSpan.FromMinutes(1)
+                : expiry;
+            await Transistion(lastSerialisedState, serialsied, stateExpiry);
+        }
+        private async Task<string> Transistion(string fromState, string toState, TimeSpan? newStateExpiry = null)
+        {
+            var setCommand = newStateExpiry == null ? $"redis.call(\'set\',\'{key}\',\'{toState}\')" : $"redis.call(\'setex\',\'{key}\', {newStateExpiry.Value.TotalSeconds},\'{toState}\' )";
+            var script = $"local prev = redis.call(\'get\', \'{key}\') if not(prev) or prev == \'{fromState}\' then {setCommand} return \'{toState}\' else return prev end";
+            var execution = (await db.ScriptEvaluateAsync(script))?.ToString();
+            return execution;
+        }
+        private async Task<CircuitBreakerStateDescriptor> ReadDescriptor()
+        {
+            var desc = new CircuitBreakerStateDescriptor(CircuitBreakerState.Closed, Clock.Now(),
+                TimeSpan.FromMinutes(1));
+            var src = await db.StringGetAsync(key);
+
+            if (!src.IsNullOrEmpty && !IsNullOrWhiteSpace(src))
+            {
+                desc = JsonConvert.DeserializeObject<CircuitBreakerStateDescriptor>(src, JsonSerializationSettings);
+            }
+
+            return desc;
+        }
+
+        public async Task Initialize(ISubscriber subscriber)
+        {
+            keySpaceObserver = new KeySpaceObserver(dbId, key, subscriber);
+            keySpaceSubscription = keySpaceObserver.Subscribe(KeySpaceNotificationHandler);
+            await keySpaceObserver.Initialize();
+        }
+
+        private void KeySpaceNotificationHandler(string target, string operation)
+        {
+            if (key != target)
+            {
+                return;
+
+            }
+            switch (operation)
+            {
+                case "expired":
+                {
+                    Transistion(
+                            null,
+                            JsonConvert.SerializeObject(new CircuitBreakerStateDescriptor(CircuitBreakerState.HalfOpen, Clock.Current.Now()), JsonSerializationSettings))
+                        .Wait();
```

Comment:
This wait is a potential deadlock. Is there a way to use an async handler with the Redis library? Otherwise, this should be called within a `Task.Run`.